src/devices_models/device_constructors/common/constructor_validations.jl

```diff
@@ -19,12 +20,15 @@ function validate_available_services(
 
     services_mapping = PSY.get_contributing_device_mapping(sys)
     for s in services_
-        if !isempty(services_mapping[(
-            type = S,
-            name = PSY.get_name(s),
-        )].contributing_devices)
+        contributing_devices_ =
+            services_mapping[(type = S, name = PSY.get_name(s))].contributing_devices
+        contributing_devices =
+            [d for d in contributing_devices_ if typeof(d) ∉ incompatible_device_types]
+        if !isempty(contributing_devices)
             push!(services, s)
         end
+        continue
+        @warn("The contributing devices for service $(PSY.get_name(service)) is empty, consider removing the service from the system")
     end
```

Comment:
I think you want something like this:

```suggestion
        if isempty(contributing_devices)
             @warn("The contributing devices for service $(PSY.get_name(service)) is empty, consider removing the service from the system")
        else
            push!(services, s)
        end
    end
```